src/jsrs_parser.cc

```diff
@@ -74,28 +76,30 @@ static constexpr Local<Value> (*kParseFunctions[])(Isolate*,
 };
 
 Local<Value> Parse(Isolate* isolate, const String::Utf8Value& in) {
-  size_t size;
-  const char* to_parse =
-      internal::PrepareString(isolate, *in, in.length(), &size);
-  if (!to_parse) {
-    return Undefined(isolate);
-  }
+  const char* str = *in;
+  const size_t length = in.length();
+  const char* end = str + length;
 
   Type type;
-  if (!GetType(to_parse, to_parse + size, &type)) {
+
+  size_t start_pos = internal::SkipToNextToken(str, end);
```

Comment:
I investigated this topic and found out that the main difference between `size_t` and `ptrdiff_t` is that former is unsigned integer, whereas latter is signed.
Also, there are a lot of arguments on this topic and generally it is recommended to use signed types for array indexing as well as pointer arithmetic.
At the same time Google C++ Style Guide says:

> When appropriate, you are welcome to use standard types like size_t and ptrdiff_t.
> 
> We use int very often, for integers we know are not going to be too big, e.g., loop counters. Use plain old int for such things. You should assume that an int is at least 32 bits, but don't assume that it has more than 32 bits. If you need a 64-bit integer type, use int64_t or uint64_t.

I guess it will be better to substitute almost all `size_t` uses with `ptrdiff_t` or plain `int`, but we need to discuss which one will suit us better.
Also I think we need new issue and corresponding PR for this topic, so that we can talk it over and weigh all the pros and cons.